third_party/nvfuser/csrc/scheduler/normalization.cpp

```diff
@@ -944,9 +1199,22 @@ std::shared_ptr<ReductionParams> persistentHeuristic(
     const size_t max_input_dtype_size,
     const int64_t max_persistent_buffer_size,
     size_t vectorize_factor,
-    bool project_persistent_buffers) {
+    bool project_persistent_buffers,
+    const bool combined_inner_outer_reduction) {
   std::shared_ptr<ReductionParams> rparams;
-  if (fastest_dim_reduction) {
+  if (combined_inner_outer_reduction) {
+    const int64_t outer_dim_numel = total_iteration_numel;
+    const int64_t inner_dim_numel = inner_most_dimension_numel;
+    // If the input is a half, we need to use 4 bytes for the temporary buffer
+    const size_t tmp_gmem_dtype_size =
+        max_input_dtype_size == 2 ? 4 : max_input_dtype_size;
```

Comment:
This seems strange as it effectively means the data type size of the temporary buffer should be equal to the maximum input data type size, except when the the maximum input size is 2. I understand this is true and sufficient for the case we are evaluating, but is there anything that guarantees this relationship? For example, I think it is possible to define a fusion that has a double input tensor, which is then cast to float before any use, and follow the inner-outer normalization pattern. In this case the temporary buffer should be float, but the max input size is 8. This is a rather synthetic case, but why not just grabbing the temporary buffer TensorView and finding out its data type size?

Reply:
The temporary buffer TensorView is NOT exist yet, it will be created after we have a schedule. AFAIK, If the input is float/double, then the code will use float/double to do the calculation. I changed to use `dataTypeSize(outer_reduction_tvs[0]->getDataType().value())`, it should conver the case you mensioned.

Reply:
> AFAIK, If the input is float/double, then the code will use float/double to do the calculation. 

It isn't necessarily the case. It's up to the user. NVFuser doesn't just provide prefix APIs like a layer-norm forward and backward, but we define more primitive operations and the user is free to define whatever composite operations. The synthetic example I mentioned above is that it is possible for the user to define a computation that is similar to the layer-norm backward but with the particular data-type pattern. Since it's not rejected by this scheduler, assuming the input type has the same size as the temporary buffer is incorrect.

>  I changed to use dataTypeSize(outer_reduction_tvs[0]->getDataType().value()), it should cover the case you mensioned.

👍